Library/Formula/websockify.rb

```diff
@@ -0,0 +1,70 @@
+require "FileUtils"
+
+class Websockify < Formula
+  homepage "https://github.com/kanaka/websockify"
+  url "https://github.com/kanaka/websockify/archive/v0.6.0.tar.gz"
+  sha1 "f2bffebc9e4d94ad3a329638bee63e899eaad9c0"
+
+  depends_on :python if MacOS.version <= :snow_leopard
+
+  def install
+    system "make"
+
+    Dir.chdir("other") do
+      system "make"
+    end
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+    bin.install Dir["#{libexec}/bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+
+    # Move this so it is not installed with the other docs
+    move 'docs/websockify.1', 'websockify.1'
+    (share+"man"+"man1").install 'websockify.1'
+    (doc+"websockify").install 'docs'
+
+    # Upstream Bug: Not named '.dylib'
+    lib.install 'rebind.so'
+
+    # Install C version of websockify
+    move 'other/websockify', 'other/websockify-c'
+    bin.install 'other/websockify-c'
+
+    # Clean up before installing client
+    Dir.mkdir 'client'
+    move 'wsirc.html', 'client/wsirc.html'
+    move 'wstelnet.html', 'client/wstelnet.html'
+    rmdir 'include/web-socket-js-project'
```

Comment:
Understood. The build won't fail for version 0.6.0 but will fail if in the
future there's a new release, this folder is non-empty and the maintainer
of the recipe pushes it to master... ie think of this as if an assert.
On 8 Feb 2015 12:14, "Dominyk Tiller" <notifications@github.com> wrote:

> In Library/Formula/websockify.rb
> <https://github.com/Homebrew/homebrew/pull/36643#discussion_r24296341>:
>
> > +    move 'docs/websockify.1', 'websockify.1'
> > +    (share+"man"+"man1").install 'websockify.1'
> > +    (doc+"websockify").install 'docs'
> > +
> > +    # Upstream Bug: Not named '.dylib'
> > +    lib.install 'rebind.so'
> > +
> > +    # Install C version of websockify
> > +    move 'other/websockify', 'other/websockify-c'
> > +    bin.install 'other/websockify-c'
> > +
> > +    # Clean up before installing client
> > +    Dir.mkdir 'client'
> > +    move 'wsirc.html', 'client/wsirc.html'
> > +    move 'wstelnet.html', 'client/wstelnet.html'
> > +    rmdir 'include/web-socket-js-project'
>
> We can't really make the build fail. Builds failing = lots of issues
> complaining get opened. It needs to fail non-fatally. Tim or one of the
> other maintainers can make the call on this one though - They may see the
> need for rmdir.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/36643/files#r24296341>.
>